### ISSUE TITLE
[Question]: Which epoch to choose?

### ISSUE THREAD

**Issue by IndustrialOne** [IAHispano/Applio#866]
![appliovoicetraining](https://github.com/user-attachments/assets/f4697b95-f896-49cd-9460-10bb0cf2646e)

This is my training data (200 epochs total) with a 5.5 minute dataset. Which .PTH at which epoch should I save? I have no idea what I'm looking at on these charts.

I was told to pay attention to node collapses on the loss/g/total chart, but that happens from the very start so I guess I should look at the middle where it starts sloping? Should I keep the .PTH at step 3000 right before the decline in the middle?

Also, in general, how many epochs per minute of audio should I choose? I need some parameters.

**Comment by MalcolmReed-ent** [comment]
Save the model around epoch 2000-2500 where the generator loss (loss/g/total) has stabilized and before the decline after epoch 3000. For a 5.5-minute dataset, your total of 200 epochs is appropriate, as a general rule is to use about 40-50 epochs per minute of audio.

**Comment by IndustrialOne** [comment]
> Save the model around epoch 2000-2500 where the generator loss (loss/g/total) has stabilized and before the decline after epoch 3000. For a 5.5-minute dataset, your total of 200 epochs is appropriate, as a general rule is to use about 40-50 epochs per minute of audio.

I forget how many steps per epoch this test was but if 5500 steps is 200 epochs that would be 27.5/epoch.

You told me to save at 2500k before the first slope which would be around 90 epochs. This contradicts other parts of your advice.
